src/test/java/com/wikia/webdriver/testcases/discussions/CreatingReplyTests.java

```diff
@@ -8,73 +8,185 @@
+    int expandedEditorHeight = replyCreator.getEditorHeight();
+    Assertion.assertTrue(expandedEditorHeight > originalEditorHeight, "Reply editor should expand.");
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
   }
 
-  private void userOnMobileMustBeLoggedInToUseReplyCreator() {
-    ReplyCreatorMobile replyCreator = new PostDetailsPage().openDefaultPost().getReplyCreatorMobile();
+  @Test(groups = USER_ON_DESKTOP_GROUP)
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DiscussionsConstants.DESKTOP_RESOLUTION)
+  public void userOnDesktopCanCreateReplyOnPostDetailsPage() {
+    final PostDetailsPage page = new PostDetailsPage().open(createPostAsUserRemotely().getId());
+    final ReplyCreator replyCreator = page.getReplyCreatorDesktop();
 
+    assertThatUserCanCreateReply(page, replyCreator);
+  }
+
+  @Test(groups = USER_ON_DESKTOP_GROUP)
+  @Execute(asUser = User.USER_2)
+  @InBrowser(browser = Browser.FIREFOX, browserSize = DiscussionsConstants.DESKTOP_RESOLUTION)
+  public void userOnDesktopShouldFollowPostWhenAddedReplyOnPostDetailsPage() {
+    final PostDetailsPage page = new PostDetailsPage().open(createPostAsUserRemotely().getId());
+    final ReplyCreator replyCreator = page.getReplyCreatorDesktop();
+
+    assertThatPostIsFollowedWhenReplyAdded(page, replyCreator);
+    assertThatPostCanBeUnfollowed(page);
+  }
+
+  // Testing methods
+
+  private void userOnDesktopMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.REGISTER_PAGE);
+  }
+
+  private void anonymousUserOnReplyEditorClickIsRedirectedTo(final ReplyCreator replyCreator, final String urlFragment) {
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickOkButtonInSignInDialog();
-
     Assertion.assertTrue(replyCreator.click().isModalDialogVisible());
 
     replyCreator.clickSignInButtonInSignInDialog();
+    Assertion.assertTrue(driver.getCurrentUrl().contains(urlFragment));
+  }
+
+  private void userOnMobileMustBeLoggedInToUseReplyCreator(final ReplyCreator replyCreator) {
+    anonymousUserOnReplyEditorClickIsRedirectedTo(replyCreator, MercurySubpages.JOIN_PAGE);
+  }
+
+  private PostEntity.Data createPostAsUserRemotely() {
+    return DiscussionsOperations.using(User.USER, driver).cratePostWithUniqueData();
+  }
+
+  private void assertThatUserCanCreateReply(PostDetailsPage page, ReplyCreator replyCreator) {
+    replyCreator.click().clickGuidelinesReadButton();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
+    replyCreator.add(TextGenerator.createUniqueText());
+    Assertion.assertTrue(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_ACTIVE_MESSAGE);
+
+    replyCreator.clearText();
+    Assertion.assertFalse(replyCreator.isSubmitButtonActive(), SUBMIT_BUTTON_INACTIVE_MESSAGE);
+
```

Comment:
Those steps are not redundant there were described here: SOC-2556. I agree that ideally they should be in separate test, but taking time into consideration and test suite (SOC-2556) I'll leave it as it is.

Those tests check only one user for mobile and desktop (reply editors differ for those two devices).